pythonforandroid/recipes/android/src/android/_android.pyx

```diff
@@ -280,17 +280,20 @@ class AndroidBrowser(object):
 import webbrowser
 webbrowser.register('android', AndroidBrowser)
 
-cdef extern void android_start_service(char *, char *, char *)
-def start_service(title=None, description=None, arg=None):
-    cdef char *j_title = NULL
-    cdef char *j_description = NULL
-    if title is not None:
-        j_title = <bytes>title
-    if description is not None:
-        j_description = <bytes>description
-    if arg is not None:
-        j_arg = <bytes>arg
-    android_start_service(j_title, j_description, j_arg)
+
+def start_service(title=None, description=None, arg=None,
+                  as_foreground=True):
+    from jnius import autoclass
+    mActivity = autoclass('org.kivy.android.PythonActivity').mActivity
+    if as_foreground:
+        mActivity.start_service(
+            title or "Background Service", description or "", arg or ""
+        )
+    else:
+        mActivity.start_service_not_as_foreground(
+            title or "Background Service", description or "", arg or ""
+        )
```

Comment:
why don't you make these default function param then?
```python
def start_service(
    title="Background Service",
    description="",
    arg="",
    as_foreground=True):
```

Reply:
good idea! I did so now, but added some extra code to translate `None` to the new defaults - someone might have read the old signature and passed `None` explicitly, and I don't want to break someone's old code

Reply:
I didn't think about this case, who cares about retro compatibility in the Python world :sweat_smile: